packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/http/invoke.ts

```diff
@@ -0,0 +1,180 @@
+import { Construct } from 'constructs';
+import * as events from '../../../aws-events';
+import * as iam from '../../../aws-iam';
+import * as sfn from '../../../aws-stepfunctions';
+import { integrationResourceArn } from '../private/task-utils';
+
+/**
+ * The style used when applying URL encoding to array values.
+ *
+ */
+export enum ArrayEncodingFormat {
+  /**
+   * Encode arrays using brackets. For example, {"array": ["a","b","c"]} encodes to "array[]=a&array[]=b&array[]=c"
+   */
+  BRACKETS = 'BRACKETS',
+  /**
+   * Encode arrays using commas. For example, {"array": ["a","b","c"]} encodes to "array=a,b,c,d"
+   */
+  COMMAS = 'COMMAS',
+  /**
+   * Encode arrays using the index value. For example, {"array": ["a","b","c"]} encodes to "array[0]=a&array[1]=b&array[2]=c"
+   */
+  INDICES = 'INDICES',
+  /**
+   * Repeat key for each item in the array. For example, {"array": ["a","b","c"]} encodes to "array[]=a&array[]=b&array[]=c"
+   */
+  REPEAT = 'REPEAT',
+}
+
+/**
+ * Properties for calling an external HTTP endpoint with HttpInvoke.
+ */
+export interface HttpInvokeProps extends sfn.TaskStateBaseProps {
+  /**
+   * Permissions are granted to call all resources under this path.
+   * @example "https://api.example.com"
+   *
+   */
+  readonly apiRoot: string;
+
+  /**
+   * The API endpoint to call, relative to `apiRoot`.
+   * @example sfn.TaskInput.fromText("path/to/resource")
+   *
+   */
+  readonly apiEndpoint: sfn.TaskInput;
+
+  /**
+   * The HTTP method to use.
+   * @example sfn.TaskInput.fromText("GET")
+   *
+   */
+  readonly method: sfn.TaskInput;
+
+  /**
+   * The EventBridge Connection to use for authentication.
+   *
+   */
+  readonly connection: events.IConnection;
+
+  /**
+   * The body to send to the HTTP endpoint.
+   *
+   * @default - No body is sent with the request.
+   */
+  readonly body?: sfn.TaskInput;
+
+  /**
+   * The headers to send to the HTTP endpoint.
+   *
+   * @default - No additional headers are added to the request.
+   * @example sfn.TaskInput.fromObject({ 'Content-Type': 'application/json' })
+   */
+  readonly headers?: sfn.TaskInput;
+
+  /**
+   * The query string parameters to send to the HTTP endpoint.
+   *
+   * @default - No query string parameters are sent in the request.
+   */
+  readonly queryStringParameters?: sfn.TaskInput;
+
+  /**
+   * When `true`, the HTTP request body is the URL-encoded form data of the `RequestBody` field.
+   * When `false` (default), the HTTP request body is the JSON-serialized `RequestBody` field.
+   * If set to `true`, also sets 'content-type' header to 'application/x-www-form-urlencoded'.
+   *
+   * @default - No encoding.
+   */
+  readonly urlEncodeBody?: boolean;
+
+  /**
+   * The format of the array encoding.
+   * Only used if `urlEncodeBody` is `true`.
+   *
+   * @default - ArrayEncodingFormat.INDICES
+   */
+  readonly arrayEncodingFormat?: ArrayEncodingFormat;
```

Comment:
Given that this field is reliant on `urlEncodeBody` being set, these two fields should probably be one field. Maybe the only field here should still be called `urlEncodeBody` but have it take an `Encoding` (not sure about this name) class with a static function that has an optional input of `ArrayEncodingFormat`. Something along the lines of 
```
urlEncodeBody: Encoding.set(ArrayEncodingFormat.INDICES);
```
to include encoding format, or 
```
urlEncodeBody: Encoding.set();
```
to just include encoding with the default format.

This way, if they leave off this prop altogether, it's defaulted to `false`.

Reply:
I'll also add the note here that `set` probably isn't a great function name here.

Reply:
Thanks for pointing this out, I agree it's a bit awkward with the 2 params. I played around a bit, and I think a single prop can work with a couple extra enum options and renaming it to `urlEncodingFormat`. You can not pass it or specify `URLEncodingFormat.NONE` to get no encoding, pass it as `URLEncodingFormat.DEFAULT` to get encoding with the default array encoding (the underlying service defaults to `INDICES`), or specify the existing values to use those.